### PR TITLE
Fix heap-buffer-overflow in try_walkthrough_casetbl()

### PR DESCRIPTION
**Pull request by GustavoLCR**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Test plan**

Tests pass

### CODE REVIEW

**Comment by XVilka** [comment]
Not sure if this test failure on OpenBSD is related:
```
[XX] db/anal/arm arm-or-thumb visual bug
R2_NOPLUGINS=1 radare2 -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc 'e scr.interactive=1
e scr.null=1
Vprdfq
e scr.null=0
afi~^size[1]
' bins/mach0/arm-or-thumb
-- stdout
@@ -1 +0,0 @@
-32

-- exit status: -1
```

See https://builds.sr.ht/~xvilka/job/275595#task-test-0

**Comment by GustavoLCR** [comment]
> Not sure if this test failure on OpenBSD is related:
> 
> ```
> [XX] db/anal/arm arm-or-thumb visual bug
> R2_NOPLUGINS=1 radare2 -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc 'e scr.interactive=1
> e scr.null=1
> Vprdfq
> e scr.null=0
> afi~^size[1]
> ' bins/mach0/arm-or-thumb
> -- stdout
> @@ -1 +0,0 @@
> -32
> 
> -- exit status: -1
> ```
> 
> See https://builds.sr.ht/~xvilka/job/275595#task-test-0

I think this gets fixed with #17425

**Comment by XVilka** [comment]
It passed and merged, please rebase.

**Comment by XVilka** [comment]
As a side note - we should refactor `fcn_recurse()`, the amount of nested levels is crazy...

**Comment by trufae** [comment]
fcn_recurse should not be recursive, some bins result in stack exhaustion

File: libr/anal/jmptbl.c [hunk 1]
```diff
@@ -64,18 +64,24 @@ R_API bool try_walkthrough_casetbl(RAnal *anal, RAnalFunction *fcn, RAnalBlock *
...
+		free (jmptbl);
+		free (casetbl);
 		return false;
 	}
-	anal->iob.read_at (anal->iob.io, jmptbl_loc, jmptbl, jmptbl_size * sz);
-	anal->iob.read_at (anal->iob.io, casetbl_loc, casetbl, jmptbl_size);
 	for (case_idx = 0; case_idx < jmptbl_size; case_idx++) {
 		jmpptr_idx = casetbl[case_idx];
 
+		if (jmpptr_idx > jmptbl_size) {
```
Comment (trufae) [inline libr/anal/jmptbl.c:80]: shouldnt this be >= ?

**Comment by XVilka** [comment]
Maybe `fcn_recurse` can be made tail-recursive? So it will not result into the stack exhaustion?
- http://web.mit.edu/pmitros/www/ceti/tail-recursion-in-c.pdf

**Pull request merged**